src/configuration/index.js

```diff
@@ -0,0 +1,205 @@
+import Promise from 'pinkie';
+import { fsObjectExists, readFile } from '../utils/promisified-functions';
+import Option from './option';
+import optionSource from './option-source';
+import { cloneDeep, castArray } from 'lodash';
+import { ensureOptionValue as ensureSslOptionValue } from '../utils/parse-ssl-options';
+import OPTION_NAMES from './option-names';
+import { optionValueToRegExp } from './option-conversion';
+import createFilterFn from '../utils/create-filter-fn';
+import resolvePathRelativelyCwd from '../utils/resolve-path-relatively-cwd';
+
+const CONFIGURATION_FILENAME = '.testcaferc.json';
+
+const ERR_READ_CONFIG_FILE                 = 'An error is occurred during reading the configuration filename.';
```

Comment:
It's a `GeneralError`, isn't it?

Reply:
It's an error about problem with reading configuration file. Not just reading file.
We will convert `ERR_READ_CONFIG_FILE` to the `GeneralError` when another reading file code will be appeared.